plugin.cpp

```diff
@@ -308,21 +308,77 @@ PLUGIN_HANDLE plugin_init(ConfigCategory* config,
  *  Retrieve AssetAction object for given assetName using regular expression match for assetname in the assetActionMap
  *  @param assetActionMap
  *  @param assetName
- *  @return Return AssetAction object
+ *  @return Return collection of the asset name and associated AssetAction object
  */
-AssetAction getAssetAction(const std::map<std::string, AssetAction>& assetActionMap, const std::string& assetName)
+
+std::vector<std::pair<std::string, AssetAction>> getAssetAction(const std::vector<std::pair<std::string, AssetAction>>& assetActionMap, std::string& assetName)
 {
-	AssetAction noMatch;
-	noMatch.actn = action::DEFAULT_ACTION;
-	for (const auto& entry : assetActionMap)
+    int order = 0;
+
+	// Hold the asset with rule order as per the rules JSON to ensure ordering of rules execution
+	std::vector<std::pair<std::string,int>> activeAssets;
+	activeAssets.push_back({assetName, order});
+	std::vector<std::pair<std::string, AssetAction>> matchedAction;
```

Comment:
queue type make more sense here ..

Reply:
vector is a simple one for a small collection. we will loose the benefit of using iterator with the queue.